src/bootstrap/doc.rs

```diff
@@ -133,6 +133,49 @@ impl Step for UnstableBook {
 }
 
 #[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct CargoBook {
+    target: Interned<String>,
+    name: Interned<String>,
+    src: Interned<PathBuf>,
+}
+
+impl Step for CargoBook {
+    type Output = ();
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun) -> ShouldRun {
+        let builder = run.builder;
+        run.path("src/tools/cargo/src/doc/book").default_condition(builder.build.config.docs)
+    }
+
+    fn make_run(run: RunConfig) {
+        run.builder.ensure(CargoBook {
+            target: run.target,
+            name: INTERNER.intern_str("cargo"),
+            src: INTERNER.intern_path(PathBuf::from("src/tools/cargo/src/doc/book")),
```

Comment:
I think this and the `name` field above are best left as local variables in `run` perhaps because I don't think they're inputs so much as calculated.

Reply:
ah makes sense, I did a bit TOO much cargo culting.

I'll update this after we decide what the name should be, and then :shipit: